packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1063,16 +1333,132 @@ export class TransactionController extends BaseControllerV1<
+      return;
+    }
+    const networkClient = this.getNetworkClientById(networkClientId);
+    const { chainId } = networkClient.configuration;
+
+    let etherscanRemoteTransactionSource =
+      this.etherscanRemoteTransactionSourcesMap.get(chainId);
+    if (!etherscanRemoteTransactionSource) {
+      etherscanRemoteTransactionSource = new EtherscanRemoteTransactionSource({
+        includeTokenTransfers:
+          this.incomingTransactionOptions.includeTokenTransfers,
+      });
+      this.etherscanRemoteTransactionSourcesMap.set(
+        chainId,
+        etherscanRemoteTransactionSource,
+      );
+    }
+
+    if (!this.nonceMutexByChainId.get(chainId)) {
+      this.nonceMutexByChainId.set(chainId, new Mutex());
+    }
+    const ethQuery = new EthQuery(networkClient.provider);
+
+    const nonceTracker = new NonceTracker({
+      // TODO: Replace `any` with type
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      provider: networkClient.provider as any,
+      blockTracker: networkClient.blockTracker,
+      getPendingTransactions: this.getNonceTrackerPendingTransactions.bind(
+        this,
+        chainId,
+      ),
+      getConfirmedTransactions: this.getNonceTrackerTransactions.bind(
+        this,
+        TransactionStatus.confirmed,
+      ),
+    });
+
+    const incomingTransactionHelper = new IncomingTransactionHelper({
+      blockTracker: networkClient.blockTracker,
+      getCurrentAccount: this.getSelectedAddress,
+      getLastFetchedBlockNumbers: () => this.state.lastFetchedBlockNumbers,
+      getChainId: () => chainId,
+      isEnabled: this.incomingTransactionOptions.isEnabled,
+      queryEntireHistory: this.incomingTransactionOptions.queryEntireHistory,
+      remoteTransactionSource: etherscanRemoteTransactionSource,
+      transactionLimit: this.config.txHistoryLimit,
+      updateTransactions: this.incomingTransactionOptions.updateTransactions,
+    });
+
+    this.addIncomingTransactionHelperListeners(incomingTransactionHelper);
+
+    const pendingTransactionTracker = new PendingTransactionTracker({
+      approveTransaction: this.approveTransaction.bind(this),
+      blockTracker: networkClient.blockTracker,
+      getChainId: () => networkClient.configuration.chainId,
+      getEthQuery: () => ethQuery,
+      getTransactions: () => this.state.transactions,
+      isResubmitEnabled: this.pendingTransactionOptions.isResubmitEnabled,
+      nonceTracker,
+      publishTransaction: this.publishTransaction.bind(this),
+      hooks: {
+        beforeCheckPendingTransaction:
+          this.beforeCheckPendingTransaction.bind(this),
+        beforePublish: this.beforePublish.bind(this),
+      },
+    });
+
+    this.addPendingTransactionTrackerListeners(pendingTransactionTracker);
+
+    this.trackingMap.set(networkClientId, {
+      nonceTracker,
+      incomingTransactionHelper,
+      pendingTransactionTracker,
+    });
+
+    this.hub.emit('tracking-map-add', networkClientId);
```

Comment:
Do these events service an external purpose?

Reply:
no external use. they are used for testing/debugging